pkg/controller/datavolume-controller.go

```diff
@@ -600,24 +557,201 @@ func (r *DatavolumeReconciler) reconcileSmartClonePvc(log logr.Logger, datavolum
 				return reconcile.Result{}, err
 			}
 		} else if tmpPVC.Annotations[AnnCloneOf] == "true" {
-			done, err := r.expand(log, datavolume, tmpPVC, pvcSpec)
-			if err != nil {
-				return reconcile.Result{}, err
-			}
-			if !done {
-				return reconcile.Result{},
-					r.updateSmartCloneStatusPhase(cdiv1.ExpansionInProgress, datavolume, nil)
-			}
+			return r.expandTmpPvcAfterClone(log, datavolume, pvcSpec, tmpPVC, CsiClone)
+		}
+	}
+
+	if datavolume.Status.Phase == cdiv1.NamespaceTransferInProgress {
+		return reconcile.Result{}, nil
+	}
+
+	sourcePvcNs := datavolume.Spec.Source.PVC.Namespace
+	if sourcePvcNs == "" {
+		sourcePvcNs = datavolume.Namespace
+	}
+	r.log.V(3).Info("CSI-Clone is available")
+
+	// Get source pvc
+	sourcePvc := &corev1.PersistentVolumeClaim{}
+	if err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: sourcePvcNs, Name: datavolume.Spec.Source.PVC.Name}, sourcePvc); err != nil {
+		if k8serrors.IsNotFound(err) {
+			log.V(3).Info("Source PVC no longer exists")
+			return reconcile.Result{}, err
+		}
+		return reconcile.Result{}, err
+	}
+
+	inUse, err := r.sourceInUse(datavolume, CSICloneSourceInUse)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	populated, err := r.isSourcePVCPopulated(datavolume)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	if inUse || !populated {
+		return reconcile.Result{Requeue: true},
+			r.updateCloneStatusPhase(cdiv1.CloneScheduled, datavolume, nil, CsiClone)
+	}
+
+	// here use Temp PVC!!!
+	log.Info("Creating PVC for datavolume")
+	cloneTargetPvc, err := r.newVolumeClonePVC(datavolume, sourcePvc, pvcSpec, pvcName)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	pvc := &corev1.PersistentVolumeClaim{}
+	if err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: cloneTargetPvc.Namespace, Name: cloneTargetPvc.Name}, pvc); err != nil {
```

Comment:
this makes sense, I'll do it

Reply:
Now I think I'll leave it like it is now:

Currently it does:
get:
- notExists -> create
- exists -> check for name clash

New way would be like this, and probably harder to cover all possibilities:
create:
- created - ok
- alreadyExists:
 -- get:
    -exists -> check for name clash
    - *not exists (low probability but actually might happen)*

@awels  What do you think

Reply:
I was trying to figure out why you would need to get the PVC again, since you have everything you need from the PVC definition you attempted to created, but then I noticed in the else you are checking the pvc.Status.Phase, which you can only get from querying the PVC. For this reason I am okay with leaving it as is.